tracer/src/Datadog.Trace/Util/Http/QueryStringObfuscation/ObfuscatorFactory.cs

```diff
@@ -10,9 +10,24 @@ namespace Datadog.Trace.Util.Http.QueryStringObfuscation
 {
     internal class ObfuscatorFactory
     {
+        private static readonly IDatadogLogger Log = DatadogLogging.GetLoggerFor<ObfuscatorFactory>();
+
         internal static ObfuscatorBase GetObfuscator(double timeoutInMs, string pattern, IDatadogLogger logger, bool reportQueryString = true)
-            => string.IsNullOrEmpty(pattern) || !reportQueryString
-                   ? new NullObfuscator()
-                   : new Obfuscator(pattern, TimeSpan.FromMilliseconds(timeoutInMs), logger ?? DatadogLogging.GetLoggerFor(typeof(QueryStringManager)));
+        {
+            if (string.IsNullOrEmpty(pattern) || !reportQueryString)
```

Comment:
it was a mistake from before, but `if !reportQueryString`, then maybe we want to return the new `RedactAllObfuscator` instead? 
this bug shouldn't happen as in the `QueryStringManager` there's a guard for `!reportQueryString` but it's still weird that !reportQueryString leads to a non obfuscated query string..

Reply:
Fixed in cc3961402

Reply:
In fact it's in f380181ae